test/note_display.test.js

```diff
@@ -0,0 +1,530 @@
+      pendingTrash.uuid,
+      (mutator) => {
+        mutator.trashed = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeTrashed = true;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(2);
+  });
+
+  it('includeArchived off', async function () {
+    await this.createNote();
+    const pendingArchive = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingArchive.uuid,
+      (mutator) => {
+        mutator.archived = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeArchived = false;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(1);
+  });
+
+  it('includeArchived on', async function () {
+    await this.createNote();
+    const pendingArchive = await this.createNote();
+    await this.itemManager.changeItem(
+      pendingArchive.uuid,
+      (mutator) => {
+        mutator.archived = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.includeArchived = true;
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(2);
+  });
+
+  it('protectedSearchEnabled false', async function () {
+    const normal = await this.createNote();
+    await this.itemManager.changeItem(
+      normal.uuid,
+      (mutator) => {
+        mutator.protected = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.searchQuery = { query: 'world', protectedBodySearch: false };
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(0);
+  });
+
+  it('protectedSearchEnabled true', async function () {
+    const normal = await this.createNote();
+    await this.itemManager.changeItem(
+      normal.uuid,
+      (mutator) => {
+        mutator.protected = true;
+      }
+    );
+    const criteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.searchQuery = { query: 'world', protectedBodySearch: true };
+    });
+    expect(notesMatchingCriteria(criteria, this.itemManager.collection).length).to.equal(1);
+  });
+
+  it('tags', async function () {
+    const note = await this.createNote();
+    const tag = await this.createTag([note]);
+    const looseTag = await this.createTag([], 'loose');
+
+    const matchingCriteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.tags = [tag];
+    });
+    expect(notesMatchingCriteria(matchingCriteria, this.itemManager.collection).length).to.equal(1);
+
+    const nonmatchingCriteria = NotesDisplayCriteria.CreateCriteria((criteria) => {
+      criteria.tags = [looseTag];
+    });
+    expect(notesMatchingCriteria(nonmatchingCriteria, this.itemManager.collection).length).to.equal(0);
+  });
```

Comment:
Is this testing two distinct things? If so splitting it into two tests (potentially in a nested `describe` block) would help